add a new env variable to define the SCHEME

digiltd opened the pull request:
I am not sure how this affects Heroku, there is a default to HTTP which should prevent any breakages due to the KODI_SCHEME env variable not being set.

Would the only Heroku change be the setting of the env var by adding it to the `config:set` command i.e. 

`heroku config:set KODI_SCHEME='https' KODI_ADDRESS='your_ip_or_dynamic_address' KODI_PORT='kodi_port' KODI_USERNAME='kodi_username' KODI_PASSWORD='kodi_password' --app app-name-and-number`.

If that is the case I will change the readme to include it.

jingai commented:
Stupid question.. but where do I even enable HTTPS in Kodi?  http://kodi.wiki/view/JSON-RPC_API#HTTP doesn't list it as an option, nor do I see it in Kodi's configuration.

digiltd commented:
You don't :)

But if you are not running the skill locally then the KODI_ADDRESS var would need to be your dyndns address and that can be HTTPS e.g. `https://dyndns.domainname.com:33133`

Then you port forward or use a reverse proxy to direct traffic to your Kodi local ip using HTTP.

jingai commented:
Ah I gotcha.

Looks fine to me then.

digiltd commented:
Squashed commits

Also updated readme to include the KODI_SCHEME var in the Heroku command and added information about what should go into the vars.

jingai commented:
It'd be relevant to this PR if you want to add the variable for specifying the path too (KODI_JSONRPC_PATH?  Or something like that maybe).

digiltd commented:
I could, but as this PR is (hopefully) ready to merge it would be good to tick it off and I can clean up  a bit.

jingai commented:
Well, it isn't necessary, but this PR won't work if the user can't proxy on the default path for whatever reason (e.g., has another configured already).

Also, you should probably add something to the documentation describing that this is for a proxy.  It wasn't initially clear to me and I'm going to guess other users are just going to assume that Kodi supports HTTPS now and ask about it on the forums.

digiltd committed:
added vars to allow changes to scheme and to include a subpath if needed

digiltd commented:
Added env vars for KODI_SCHEME and KODI_SUBPATH and included some comments in the .env files.

Tested with AWS only, defaults are in place to not upset any installs that don't use the extra vars.

jingai commented:
Looks OK at a glance.  I'll make sure there's no issue with the defaults when I get home today.

jingai commented:
Sorry, I'm not ignoring this -- just have been coming home so late I don't have the energy to get to it.  Should be over with soon though.

jingai commented:
All merged.  Removed some trailing whitespace and shuffled a bit around, but it's the same functionally.
